### PR TITLE
[Ready]DNA sequencer now uses a popup

### PR DESCRIPTION
**Pull request by Time-Green**
The genetic sequencer will now give you only the list of mutations something has. You can then use it in-hand to select one of the mutations and view the sequence. It's on a 20 second cooldown.

Breaks people just running around, scanning a bunch of people and then using a script to get it all done at once. This change hardly obstructs people that use it the way that it was intended.

:cl:
tweak: The genetic sequencer has been reworked. Use it in-hand to view the sequence
/:cl:

### CODE REVIEW

**Comment by knacker48** [comment]
My main method of doing genetics was scanning a bunch of people in the halls at round start then going back to genetics and comparing the numbers to the mutations in order to solve the more difficult mutations. Will this PR invalidate this strategy?

**Comment by getbrecked** [comment]
> My main method of doing genetics was scanning a bunch of people in the halls at round start then going back to genetics and comparing the numbers to the mutations in order to solve the more difficult mutations. Will this PR invalidate this strategy?

You could copy/paste the data into your PDA notes or a text editor.

**Comment by Grazyn** [comment]
I wish there was another way to break the webscript, this really makes genetics much more painful to do.

**Comment by WarRage333** [comment]
Oh well it's still faster to use the old exploit to get powers

**Comment by py01** [comment]
Using the script is still going to be the quickest way to do genetics, you just need to do more copy-pastes.

**Comment by ATH1909** [comment]
But that script was what made genetics actually fun/playable for me...

Aw.

**Comment by ATH1909** [comment]
> this breaks that dude’s genetic solver script lol

That is literally the stated point of this PR.

**Comment by Time-Green** [comment]
> Using the script is still going to be the quickest way to do genetics, you just need to do more copy-pastes.

@py01 I considered it, but I’m counting on people using the script to lack the mental capacity to figure that out

**Comment by knacker48** [comment]
If they're using a script they're at least smart enough to setup and use a script. I've never used one but how hard do you think it would be for them to alter the script to work around this change?

**Comment by Tlaltecuhtli** [comment]
>If they're using a script they're at least smart enough to setup and use a script. 

you have to press save logs ingame put in a folder and click the dolphin.exe
you dont really need to be smart

File: code/game/objects/items/devices/scanners.dm [hunk 1]
```diff
@@ -679,17 +679,19 @@ GENE SCANNER
...
 	if (!M.has_trait(TRAIT_RADIMMUNE) && !M.has_trait(TRAIT_BADDNA)) //no scanning if its a husk or DNA-less Species
 		user.visible_message("<span class='notice'>[user] has analyzed [M]'s genetic sequence.</span>")
+		gene_scan(M, user)
 
-		gene_scan(M, user, src)
 	else
-
 		user.visible_message("<span class='notice'>[user] failed to analyse [M]'s genetic sequence.</span>", "<span class='warning'>[M] has no readable genetic sequence!</span>")
 
+/obj/item/sequence_scanner/attack_self(mob/user)
```
Comment (ShizCalev) [inline code/game/objects/items/devices/scanners.dm]: Need to check for tk memes

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by WaterStBlues** [comment]
Brutal. Let's see how long it takes me to make another

**Comment by Time-Green** [comment]
> Brutal. Let's see how long it takes me to make another

I was gonna hold back my trump card, but you give me no choice.
You’re going to regret saying that boy

**Comment by WaterStBlues** [comment]
>I was gonna hold back my trump card, but you give me no choice. You’re going to regret saying that boy

I won't go as far as making an autoPy script, but it'll be fun!
Instead of comparing eDick size, what if we just made the genetics scanner do what my script does, but with a cooldown between scans? Genetics has been woefully nerfed anyway.

**Comment by Time-Green** [comment]
> > I was gonna hold back my trump card, but you give me no choice. You’re going to regret saying that boy
> 
> I won't go as far as making an autoPy script, but it'll be fun!
> Instead of comparing eDick size, what if we just made the genetics scanner do what my script does, but with a cooldown between scans? Genetics has been woefully nerfed anyway.

What would prevent someone from just walking around for 10-15 minutes scanning people with the cooldown and then instantly completing all of genetics? Because it would be that easy. I’m open to suggestions, but I’m not going to make any massive changes anymore. (Feel free to do so yourself though)

**Comment by WaterStBlues** [comment]
Alright think bout this: a slowly recharging "ammo" system. A geneticist can scan four dudes right away, but then each charge after that will take 30-40 seconds a piece to recharge. This will encourage him to go back to the lab and solve more sequences before going back out and getting more of those sweet, sweet sequences.

That being said, is running around for 20 minutes, then needing to crawl back and punch in hundreds of bases any better for gameplay than just sitting at a console matching base pairs?

**Comment by Time-Green** [comment]
> Alright think bout this: a slowly recharging "ammo" system. A geneticist can scan four dudes right away, but then each charge after that will take 30-40 seconds a piece to recharge. This will encourage him to go back to the lab and solve more sequences before going back out and getting more of those sweet, sweet sequence scans.
> 
> That being said, is running around for 20 minutes, then needing to crawl back and punch in hundreds of bases any better for gameplay than just sitting at a console matching base pairs?

People are just gonna abuse it by getting more, and 4 is alot.

Yeah it’s alot better than constantly chasing people, then waiting, then chasing people. If your argument is that the solving is boring, then actually make a meaningful change to make it more engaging instead of encouraging people to exploit it.

**Comment by Time-Green** [comment]
I pretty much made a new PR. It can only show one mutations' sequence at 20 second intervals.

**Comment by WaterStBlues** [comment]
Cool - I had a few alternate ideas to run past you, will post there after
I'm done work

**Comment by 0d0be32b607ee4ecbc2358f3dd4a27de** [comment]
WAAA WAAA PEOPLE DON'T PLAY THE GAME MY WAY WAAAAAAAAA

**Comment by kopoba** [comment]
unplayble

**Comment by Time-Green** [comment]
Try using your three braincells instead of third party programs. People playing genetics legitimitely are pretty much completely unaffected.

**Comment by kopoba** [comment]
> 
> 
> Try using your three braincells instead of third party programs. People playing genetics legitimitely are pretty much completely unaffected.

like how the they unaffected?
You run like crazy around station for each sequence to find one that you are workin on and then you do this again for another one because you don't store results and cant see them.

If you try to store each of 8 mutations from scanned man you got stupid 20 seconds cooldown to just view sequence. ~160-180 seconds on one man.
then copypaste it somewhere that also takes sometime.
>People playing genetics completely unaffected.
this is a joke?

**Comment by Time-Green** [comment]
@kopoba Results are stored in the chat if that's what you mean. Just work on one mutation at a time, and if you can't solve it scan one of your monkeys and then move on.

**Comment by kopoba** [comment]
> 
> 
> @kopoba Results are stored in the chat if that's what you mean. Just work on one mutation at a time, and if you can't solve it scan one of your monkeys and then move on.

you need 20 seconds to view one result theres like 45 mutations
45*20=15 minutes to get all mutations if you lucky and have all 45 in one man in one scan
and don't even solve them
are you sure this is right?
because it seems to take ridiculous amount of time to work in genetics

**Comment by Time-Green** [comment]
In those 20 seconda you could be entering the sequence thingy. You don’t literally stand still 20 seconds

**Comment by kopoba** [comment]
> 
> 
> In those 20 seconda you could be entering the sequence thingy. You don’t literally stand still 20 seconds

So in addition you need to scan peoples again and again and again to find mutation you trying to solve.

**Comment by Time-Green** [comment]
That doesnt happen at all though. You can scan mutations to your hearts content without cooldown. Only the option to view the sequence of the specific mutation has a cooldown.

**Comment by kopoba** [comment]
> 
> 
> That doesnt happen at all though. You can scan mutations to your hearts content without cooldown. Only the option to view the sequence of the specific mutation has a cooldown.

This update make genetics worse and take alot more time and manipulation. Also  scanner don't show instability on scan.

**Comment by adamkad1** [comment]
lemme make this shiz so that it stores genes from all people scanned and you can see complete sequences with enough people scanned.

**Comment by Time-Green** [comment]
Go ahead, I don’t own the code. But that has serious balance implications and makes genetics extremely easy

**Pull request merged**